packages/engine/src/avatar/functions/moveAvatar.ts

```diff
@@ -128,9 +129,26 @@ export const updateReferenceSpaceFromAvatarMovement = (movement: Vector3) => {
 
 const _additionalMovement = new Vector3()
 
+/**
+ * Avatar movement via click/pointer position
+ */
+export const applyAutopilotInput = (entity: Entity) => {
+  const controller = getComponent(entity, AvatarControllerComponent)
+  if (!controller || controller.autopilotWalkpoint == undefined) return
+
+  const walkpoint = new Vector3()
+  walkpoint.set(controller.autopilotWalkpoint.x, controller.autopilotWalkpoint.y, controller.autopilotWalkpoint.z)
+  const avatarPos = getComponent(entity, TransformComponent).position
+  const moveDir = walkpoint.sub(avatarPos)
+
+  if (moveDir.length() > 0.5) updateLocalAvatarPosition(moveDir.normalize().multiplyScalar(0.1))
```

Comment:
we can optimize this to be `moveDir.lengthSq() > 0.5 * 0.5` and we should precompute 0.5 in a module scoped const and give it a name to avoid putting magic numbers in the code

Reply:
Will do, using square length always seems to slip my mind.